src/util/UUID.cpp

```diff
@@ -3,3 +3,72 @@
 //
 
 #include "UUID.hpp"
+#include <array>
+
+namespace spy::util {
+    bool UUID::isNull() {
+        return uuid_is_null(uuidData);
+    }
+
+    UUID UUID::generate() {
+        UUID u;
+        uuid_generate(u.uuidData);
+        return u;
+    }
+
+    void UUID::clear() {
+        uuid_clear(uuidData);
+    }
+
+    std::string UUID::to_string() const {
+        char uuidRawString[37];
+        uuid_unparse(uuidData, uuidRawString);
+        return std::string(uuidRawString);
+    }
+
+    std::string UUID::to_string_upper() const {
+        char uuidRawString[37];
+        uuid_unparse_upper(uuidData, uuidRawString);
+        return std::string(uuidRawString);
+    }
+
+    std::string UUID::to_string_lower() const {
+        char uuidRawString[37];
+        uuid_unparse_lower(uuidData, uuidRawString);
+        return std::string(uuidRawString);
+    }
+
+    bool UUID::operator==(const UUID &rhs) const {
+        return uuid_compare(uuidData, rhs.uuidData) == 0;
+    }
+
+    bool UUID::operator<(const UUID &rhs) const {
+        return uuid_compare(uuidData, rhs.uuidData) < 0;
+    }
+
+    bool UUID::operator>(const UUID &rhs) const {
+        return rhs < *this;
+    }
+
+    bool UUID::operator<=(const UUID &rhs) const {
+        return !(rhs < *this);
```

Comment:
```suggestion
        return !(rhs > *this);
```

Reply:
Das passt schon, hier sind `rhs` und `this` vertauscht, damit direkt `operator<` verwendet wird und nicht `operator>`, der ja wiederum `operator<` aufruft

Reply:
klar, sry